web/mysql_typemap.go

```diff
@@ -0,0 +1,263 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package web
+
+import (
+	"github.com/cloudspannerecosystem/harbourbridge/internal"
+	"github.com/cloudspannerecosystem/harbourbridge/spanner/ddl"
+)
+
+// TODO(searce): Decide on the issue message that needs to be shown
+// in frontend.
+var mysqlTypeMap = map[string][]typeIssue{
+	"bool": []typeIssue{
+		typeIssue{T: ddl.Bool},
+		typeIssue{T: ddl.String, Issue: internal.Widened},
+		typeIssue{T: ddl.Int64, Issue: internal.Widened}},
+	"boolean": []typeIssue{
+		typeIssue{T: ddl.Bool},
+		typeIssue{T: ddl.String, Issue: internal.Widened},
+		typeIssue{T: ddl.Int64, Issue: internal.Widened}},
+	"varchar": []typeIssue{
+		typeIssue{T: ddl.String},
+		typeIssue{T: ddl.Bytes, Issue: internal.Widened, Brief: internal.IssueDB[internal.Widened].Brief}},
+	"char": []typeIssue{
+		typeIssue{T: ddl.String},
+		typeIssue{T: ddl.Bytes, Issue: internal.Widened}},
```

Comment:
Why is this tagged with Widened?
This tag means that the spanner type will use more storage, but choose bytes here will use the same storage won't it? Is there an encoding issue I'm overlooking?

Same comment in several places below as well.

Reply:
updated this at this and other places.